src/include/server/mir/server.h

```diff
@@ -495,7 +496,20 @@ class Server
     auto get_activation_token() const -> std::string;
 
     /// Overrides the standard set of Wayland extensions (mir::frontend::get_standard_extensions()) with a new list
+    [[deprecated("Prefer using set_wayland_extension_policy() to set a policy on a per-interface bassis.")]]
     void set_enabled_wayland_extensions(std::vector<std::string> const& extensions);
+
+    /// Set a policy for the provided wayland extension.
+    ///
+    /// Only clients whose \p policy returns `true` will be allowed to use the interface
+    /// given by \p interface_name.
```

Comment:
This phrasing seems strange: clients don't have their own policies.

```suggestion
    /// Only clients for which the \p policy returns `true` will be offered the global \p interface_name.
```